src/transformers/models/gpt_neo/modeling_gpt_neo.py

```diff
@@ -697,16 +697,28 @@ def prepare_inputs_for_generation(self, input_ids, past_key_values=None, **kwarg
             position_ids.masked_fill_(attention_mask == 0, 1)
             if past_key_values:
                 position_ids = position_ids[:, -1].unsqueeze(-1)
-
-        return {
+                
+        #if `inputs_embeds` are passed, we only want to use them in the 1st generation step
+        if inputs_embeds is not None and past_key_values is None:
+            model_inputs = {"inputs_embeds": inputs_embeds}
+        else:
+            model_inputs = {"input_ids": input_ids}
+        
+        model_inputs.update(
+            {
             "input_ids": input_ids,
             "past_key_values": past_key_values,
             "use_cache": kwargs.get("use_cache"),
             "position_ids": position_ids,
             "attention_mask": attention_mask,
             "token_type_ids": token_type_ids,
-        }
-
+            "position_ids": position_ids,
```

Comment:
I think these dict key:value pairs need one further dosage of indentation -- our CI might complain about this :)

Reply:
Fixed this in latest commit.